src/main/java/Bob.java

```diff
@@ -0,0 +1,430 @@
+            }
+        }
+
+        catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        System.out.println("File loaded.");
+        return taskList;
+    }
+
+    /*
+     * A method to save the current tasks.
+     *
+     * @parameter taskList An ArrayList of the current tasks.
+     * @parameter filePath A string representing the file path we want to save to.
+     */
+    private static void saveFile(ArrayList<Task> taskList, String filePath) {
+        try {
+            File bob = new File(filePath);
+            File data = bob.getParentFile();
+
+            if (!data.exists()) {
+                boolean directoriesCreated = data.mkdirs();
+                if (!directoriesCreated) {
+                    System.out.println("Failed to create directories.");
+                    return;
+                }
+                System.out.println("Directories created: " + data.getAbsolutePath());
+            }
+
+            if (!bob.exists()) {
+                boolean fileCreated = bob.createNewFile();
+                if (!fileCreated) {
+                    System.out.println("Failed to create file.");
+                    return;
+                }
+                System.out.println("File created: " + bob.getAbsolutePath());
+            }
+
+            FileWriter writer = new FileWriter(bob);
+            BufferedWriter bufferedWriter = new BufferedWriter(writer);
+
+            for (Task task : taskList) {
+                String taskType;
+                String line;
+                String isDone = task.getIsDone() ? "1" : "0";
+                String description = task.getDescription();
+
+                if (task instanceof Deadline) {
+                    taskType = "D";
+                    LocalDateTime deadlineDate = ((Deadline) task).getBy();
+                    String deadline = deadlineDate.format(DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm"));
+                    line = String.format("%s | %s | %s | %s", taskType, isDone, description, deadline);
+                }
+
+                else if (task instanceof Event) {
+                    taskType = "E";
+                    String from = ((Event) task).getFrom();
+                    String to = ((Event) task).getTo();
+                    line = String.format("%s | %s | %s | %s | %s", taskType, isDone, description, from, to);
+                }
+
+                else {
+                    taskType = "T";
+                    line = String.format("%s | %s | %s", taskType, isDone, description);
+                }
+
+                bufferedWriter.write(line);
+                bufferedWriter.newLine();
+            }
+            bufferedWriter.close();
+            System.out.println("File saved.");
+        }
+
+        catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+}
+
+/*
+ * This class represents a task we want to record.
+ */
+class Task {
```

Comment:
Different Classes should not be in the same .java file. Consider separating them into dedicated java files